packages/core/src/common/command.ts

```diff
@@ -268,6 +285,25 @@ export class CommandRegistry implements CommandService {
         throw new Error(`The command '${commandId}' cannot be executed. There are no active handlers available for the command.${argsMessage}`);
     }
 
+    protected async fireWillExecuteCommand(commandId: string): Promise<void> {
+        const waitables: Promise<void>[] = [];
+        this.onWillExecuteCommandEmitter.fire({
+            commandId,
+            waitUntil: (thenable: Promise<void>) => {
+                if (Object.isFrozen(waitables)) {
+                    throw new Error('waitUntil cannot be called asynchronously.');
+                }
+                waitables.push(thenable);
+            }
+        });
+        if (!waitables.length) {
+            return;
+        }
+        // Asynchronous calls to `waitUntil` should fail.
+        Object.freeze(waitables);
+        await Promise.race([Promise.all(waitables), new Promise(resolve => setTimeout(resolve, 30000))]);
```

Comment:
Wouldn't it make sense to log if something has timeouted? Correct me if I am wrong, but here we wait for the `waitables` but max 30 sec.

Reply:
> Wouldn't it make sense to log if something has timeouted?

It will log a follow-up error that the handler is not found.

I've actually aligned it with VS Code. They wait for some reasons for 30s, maybe if some extensions takes long to activate on command execution. Maybe we should remove it and wait for real issues.

Reply:
Let's leave it as is for now and adjust later if required.